### PR TITLE
ACCESS-ESM1.6 Testing

### PR DESCRIPTION
**Pull request by harshula**
This PR is only for testing.
* First test: improve mechanisms to disable generic tracers.

---
:rocket: The latest prerelease `access-esm1p6/pr15-21` at 919f58e35fbf2f0b01060b92e11c50f56faafb85 is here: https://github.com/ACCESS-NRI/ACCESS-ESM1.6/pull/15#issuecomment-2537480555 :rocket:

### CODE REVIEW

File: spack.yaml [hunk 1]
```diff
@@ -2,24 +2,29 @@
...
 # configuration settings.
+
+# Build with:
+# - UM7 from dev-access-esm1.6
+# - MOM5 from ESM1.5
+# - CICE4 from ESM1.5 and change to CICE5 once CICE5 SPR has been updated.
 spack:
   specs:
-    - access-esm1p6@git.2024.10.0
+    - access-esm1p6@git.dev_2024.12.0 ~generic-tracers
```
Comment (aidanheerdegen) [inline spack.yaml:12]: Sorry if we covered this already, but why isn't the variant specified in the `mom5` package requirements?

File: spack.yaml [hunk 2]
```diff
@@ -42,10 +47,10 @@ spack:
         - '@1.10.11'
     access-fms:
       require:
-        - '@git.dev_2024.11.0'
+        - '@git.dev_2024.12.0'
     access-generic-tracers:
       require:
-        - '@git.dev_2024.11.0'
```
Comment (aidanheerdegen) [inline spack.yaml:48]: Seems odd to me to specify a version of generic tracers when they're not being used. Not saying it's wrong, just odd.

**Review by aidanheerdegen — DISMISSED** [review]
LGTM. Just some questions/comments.

File: spack.yaml (hunk 1 above)
Comment (harshula) [reply spack.yaml:12]: The SBR contains:
```
    # TODO: Use the access-esm1.6 branch when it has been confirmed that
    #       MOM5 master supports building with generic tracers disabled.
    depends_on(
        "mom5@access-esm1.5 type=ACCESS-CM",
        type="run",
        when="~generic-tracers",
    )
    depends_on(
        "mom5@access-esm1.6 type=ACCESS-ESM",
        type="run",
        when="+generic-tracers",
    )
```

File: spack.yaml (hunk 2 above)
Comment (harshula) [reply spack.yaml:48]: During the upstream Spack meeting, `spack.packages` was described as the the spec if that package is used.

File: spack.yaml (hunk 1 above)
Comment (aidanheerdegen) [reply spack.yaml:12]: You see, I knew there was a reason.

File: spack.yaml (hunk 2 above)
Comment (aidanheerdegen) [reply spack.yaml:48]: So it was! Thanks again.

**Review by ccarouge — APPROVED** [review]
Looks good to me.

**Comment by harshula** [comment]
Hi @aidanheerdegen & @ccarouge , The relevant changes in this test PR has been merged into main via PR https://github.com/ACCESS-NRI/ACCESS-ESM1.6/pull/20 . Thanks for the review!

**Pull request closed without merging**